src/field/launch_application.c

```diff
@@ -0,0 +1,1400 @@
+    UseMailArgs *args;
+
+    if (kind == 3) {
+        args = sub_02090F38(FieldSystem_GetSaveData(fieldSystem), mailId, heapId);
+    } else {
+        args = sub_02090EC0(FieldSystem_GetSaveData(fieldSystem), kind, mailId, heapId);
+    }
+    sub_0203EFD4(fieldSystem, args);
+    return args;
+}
+
+UseMailArgs *sub_0203F050(FieldSystem *fieldSystem, Pokemon *mon, HeapID heapId) {
+    UseMailArgs *args = sub_02090F00(FieldSystem_GetSaveData(fieldSystem), mon, heapId);
+    sub_0203EFD4(fieldSystem, args);
+    return args;
+}
+
+Unk0203F074 *sub_0203F074(FieldSystem *fieldSystem, HeapID heapId) {
+    Unk0203F074 *args = AllocFromHeap(heapId, sizeof(Unk0203F074));
+    args->saveData = FieldSystem_GetSaveData(fieldSystem);
+    args->fieldSystem = fieldSystem;
+    args->unk8 = &fieldSystem->unk_10C;
+    FieldSystem_LaunchApplication(fieldSystem, &_020FA2F4, args);
+    return args;
+}
+
+void sub_0203F0A8(FieldSystem *fieldSystem, UnkStruct_ScrCmd408 *args) {
+    OVY_MGR_TEMPLATE template = _020FA2E4;
+    FieldSystem_LaunchApplication(fieldSystem, &template, args);
+}
+
+static Unk0203F0D0 *sub_0203F0D0(HeapID heapId, SaveData *saveData, int partyIdx, int *a3, int a4) {
+    Unk0203F0D0 *ptr = AllocFromHeap(heapId, sizeof(Unk0203F0D0));
+    memset(ptr, 0, sizeof(Unk0203F0D0));
+    ptr->mon = Party_GetMonByIndex(SaveArray_Party_Get(saveData), partyIdx);
+    SaveFashionData *fashionSave = Save_FashionData_Get(saveData);
+    SaveFashionDataSub *var1 = sub_0202B9B8(fashionSave, 0);
+    FashionCase *fashionCase = Save_FashionData_GetFashionCase(fashionSave);
+    ptr->unk4 = var1;
+    ptr->fashionCase = fashionCase;
+    ptr->options = Save_PlayerData_GetOptionsAddr(saveData);
+    ptr->gameStats = Save_GameStats_Get(saveData);
+    ptr->profile = Save_PlayerData_GetProfileAddr(saveData);
+    ptr->unk18 = a3;
+    ptr->unk1C = a4;
+    return ptr;
+}
+
+static BOOL sub_0203F134(TaskManager *taskman) {
+    FieldSystem *fieldSystem = TaskManager_GetFieldSystem(taskman);
+    Unk0203F134 *data = TaskManager_GetEnvironment(taskman);
+
+    switch (data->state) {
+    case 0:
+        sub_0203F1E8(fieldSystem, data->unkC);
+        data->state++;
+        break;
+    case 1:
+        if (!FieldSystem_ApplicationIsRunning(fieldSystem)) {
+            u16 *ptr = data->unk4;
+            if (ptr != NULL) {
+                if (data->unk8 == 1) {
+                    *ptr = 1;
+                } else {
+                    *ptr = 0;
+                }
+            }
+            FreeToHeap(data->unkC);
+            FreeToHeap(data);
+            return TRUE;
+        }
+        break;
+    }
+    return FALSE;
+}
+
+void sub_0203F198(TaskManager *taskman, u16 *ret, SaveData *saveData, u16 a3, u16 a4) {
+    u32 size;
+    u8 *ptr;
+
+    Unk0203F134 *data = AllocFromHeap(HEAP_ID_32, sizeof(Unk0203F134));
+
+    ptr = (u8 *)data;
+    size = sizeof(Unk0203F134);
+    while (size) {
```

Comment:
this feels like it should be inline somewhere

Reply:
Do you know which function? Couldn't find one with a quick scan

Reply:
It's memset.
`memset(data, 0, sizeof(Unk0203F134);`

Reply:
Oh right, thanks!